src/components/App.js

```diff
@@ -74,10 +85,51 @@ class App extends Component {
 
   componentDidMount() {
     verticalLayoutBreakpoint.addListener(this.onLayoutChange);
+    shortcuts.on(
+      L10N.getStr("symbolSearch.search.key2"),
+      this.toggleSymbolModal
+    );
+    shortcuts.on("Escape", this.onEscape);
   }
 
   componentWillUnmount() {
     verticalLayoutBreakpoint.removeListener(this.onLayoutChange);
+    shortcuts.off(
+      L10N.getStr("symbolSearch.search.key2"),
+      this.toggleSymbolModal
+    );
+    shortcuts.off("Escape", this.onEscape);
+  }
+
+  onEscape(shortcut, e) {
+    const { activeSearchState, closeActiveSearch } = this.props;
+
+    if (activeSearchState) {
+      e.preventDefault();
+      closeActiveSearch();
+    }
+  }
+
+  toggleSymbolModal(_, e: SyntheticEvent) {
+    const {
+      selectedSource,
+      activeSearchState,
+      closeActiveSearch,
+      setActiveSearch
+    } = this.props;
+
+    e.preventDefault();
+    e.stopPropagation();
+
+    if (!selectedSource) {
+      return;
+    }
+
+    if (activeSearchState == "symbol") {
```

Comment:
I don't understand this test here. A comment or using a constant for "symbol" might help understanding better what it does.

Reply:
yeah, we figure w/ flow we can use strings... bc they're type checked, but perhaps we should be stricter w/ constants